src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncValueTaskMethodBuilder.cs

```diff
@@ -42,38 +42,41 @@ public struct AsyncValueTaskMethodBuilder
 
         /// <summary>Associates the builder with the specified state machine.</summary>
         /// <param name="stateMachine">The state machine instance to associate with the builder.</param>
-        public void SetStateMachine(IAsyncStateMachine stateMachine) => _methodBuilder.SetStateMachine(stateMachine);
+        public void SetStateMachine(IAsyncStateMachine stateMachine)
+            => AsyncMethodBuilderCore.SetStateMachine(stateMachine, m_task);
 
         /// <summary>Marks the task as successfully completed.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void SetResult()
         {
-            if (_useBuilder)
+            if (m_task is null)
             {
-                _methodBuilder.SetResult();
+                m_task = System.Threading.Tasks.Task.s_cachedCompleted;
             }
             else
             {
-                _haveResult = true;
+                AsyncMethodBuilderCore.SetExistingTaskResult(m_task, default);
             }
         }
 
         /// <summary>Marks the task as failed and binds the specified exception to the task.</summary>
         /// <param name="exception">The exception to bind to the task.</param>
-        public void SetException(Exception exception) => _methodBuilder.SetException(exception);
+        public void SetException(Exception exception)
+            => AsyncMethodBuilderCore.SetException(ref m_task!, exception); // TODO-NULLABLE-GENERIC
 
         /// <summary>Gets the task for this builder.</summary>
         public ValueTask Task
         {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Was it previously getting inlined?  Even if it was, isn't this now increasing the size of the asm?

Reply:
Removed